source/GameMode.cpp

```diff
@@ -76,7 +79,9 @@ bool GameMode::init(const std::shared_ptr<cugl::AssetManager>& assets) {
 	playerId = net.getPlayerID();
 	gm.setPlayerId(playerId);
 	// gm.setDonuts(shipModel);
-	donutModel = donuts.at(playerId);
+	donutModel = donuts.at(static_cast<unsigned long>(playerId));
+	donutModel->setColorId(0);
```

Comment:
Maybe I'm missing something here, but why is this hard coded to 0?

Reply:
Right now the player is always hardcoded to the first color in the JSON file. This is to ensure that the player has an unique color and everyone else is assigned one of the remaining colors. I set it to zero here to ensure that field is at least a sensible number. I should mark it as a TODO to improve later, unless you have suggestions?

Reply:
I don't mind it for now as long as the players' colors are consistent across devices. Is this the case?

Reply:
Player's color depends on the playerId

Reply:
Wait, I'm confused. So is the player's color their player ID, or is it 0?

Reply:
Player's color depends on their playerId but is not equal to it. ColorId is the idx into the playerColor vector. Device client has "yellow" hardcoded into JSON file, so its colorId is 0(yellow). Everyone else gets a number that is NOT 0 calculated using % with their playerId.

Reply:
But isn't the whole point of the game to make each player's color consistent across devices? If I roll past a yellow breach and I'm not yellow, I need to be able to tell the person who is that they need to come fix it. How can we do that if everyone is hard-coded to yellow on their own screen?

Reply:
Good point lol